### ISSUE TITLE
Few async methods are not awaitable

### ISSUE THREAD

**Issue by sarmut** [shubhamranjan/dotnet-etcd#1]
Hi,

Methods like 'PutAsync()' are marked as 'void'.  Can this return type changed to Task and return whatever the underlying async response comes from KVClient's corresponding async methods?  This will make the caller to apply proper 'await' patterns.

Example:

**Existing:**
```
        public async void PutAsync(string key, string val)
        {
            try
            {
                await _kvClient.PutAsync(new PutRequest
                {
                    Key = ByteString.CopyFromUtf8(key),
                    Value = ByteString.CopyFromUtf8(val)
                }, _headers);
            }
            catch
            {
                ResetConnection();
                throw;
            }
       }
```

**Suggested Change:**

        public async Task<PutResponse> PutAsync(string key, string val)
        {
            try
            {
                return await _kvClient.PutAsync(new PutRequest
                {
                    Key = ByteString.CopyFromUtf8(key),
                    Value = ByteString.CopyFromUtf8(val)
                }, _headers);

            }
            catch
            {
                ResetConnection();
                throw;
            }
        }

Thank you

**Comment by shubhamranjan** [comment]
@sarmut 
I agree with the return type not being `void` for methods like `PutAsync`. I will change the same and it won't be a breaking change as well for similar return types.

However , when you say `return whatever the underlying async response comes from KVClient's corresponding async methods`, methods like `GetAsync` don't return the whole of the output but return `Task<string>`. Can you provide a specific use case when you would like to use the whole response not just the `Get` value.

**Comment by sarmut** [comment]
@shubhamranjan 
There are situations the whole response is needed, for example, logging.  I need to log the certain pieces from response header to analyze later.  So it'd be helpful not to throw away the underlying response object.

One more request.  Is 'ResetConnection()' needed?  Scenarios is that if I share this client object with multiple threads/async tasks, if one task fails then all the other tasks connected to that particular client also will fail because the underlying channel is destroyed.

**Comment by shubhamranjan** [comment]
1. Noted.I will try to provide the response object maintaining backward compatibility.
2. I am aware of this. Unable to get something like ConnectionException on disconnect, hence added that in the universal exception catch block to reset connection.If connection fails, all subsequently calls will fail.To avoid a scenario for shared objects, I will try to look for more improvements.

**Comment by shubhamranjan** [comment]
Update: Will patch this in the upcoming week.
